src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -1,9 +1,133 @@
 <script setup>
+import { ref } from 'vue';
+import { MaButton } from "@mobileaction/action-kit";
+import { MaInput } from "@mobileaction/action-kit";
+import { MaSelect } from "@mobileaction/action-kit";
+
+const nGrams = ref([]);
+const text = ref('');
+const selectedNgrams = ref([]);
+const options = [
+  { label: '1-gram', value: 1 },
+  { label: '2-gram', value: 2 },
+  { label: '3-gram', value: 3 },
+  { label: '4-gram', value: 4 },
+  { label: '5-gram', value: 5 },
+  { label: '6-gram', value: 6 },
+  { label: '7-gram', value: 7 },
+  { label: '8-gram', value: 8 },
+  { label: '9-gram', value: 9 },
+  { label: '10-gram', value: 10 },
+];
+
+const generateNGram = (text, n) => {
+  const result = new Set();
+  const words = text.split(" ");
+  for (let i = 0; i <= words.length - n; i++) {
+    const nGram = words.slice(i, i + n).join(' ');
+    result.add(nGram);
+  }
+  return Array.from(result);
+};
+
+const generateNGrams = (removeUnwanted) => {
+  if (!text.value.trim()) { return; }
+  const filteredText = removeUnwanted ? removeUnwantedWords() : text.value;
+  nGrams.value = selectedNgrams.value.map(n => ({ n, keywords: generateNGram(filteredText, n) }));
+  nGrams.value = nGrams.value.filter(nGram => nGram.keywords.length);
+};
+
+const removeUnwantedWords = () => {
```

Comment:
The removeUnwantedWords function should be a helper function inside generateNGrams, so it should be converted into a function that takes the necessary parameter and works accordingly.

const removeUnwantedWords = (inputText) => {
  const unwantedWords = new Set(['a', 'an', 'the', 'is']);
  return inputText.split(' ').filter(word => !unwantedWords.has(word.toLowerCase())).join(' ');
};